spec/Appendix A -- Field Selection.md

```diff
@@ -0,0 +1,549 @@
+# Appendix A: Specification of FieldSelectionMap Scalar
+
+## Introduction
+
+This appendix focuses on the specification of the {FieldSelectionMap} scalar type. {FieldSelectionMap} is
+designed to express semantic equivalence between arguments of a field and fields within the result 
+type. Specifically, it allows defining complex relationships between input arguments and fields in
+the output object by encapsulating these relationships within a parsable string format. It is used
+in the `@is` and `@requires` directives.
+
+To illustrate, consider a simple example from a GraphQL schema:
+
+```graphql
+type Query {
+    userById(userId: ID! @is(field: "id")): User! @lookup
+}
+```
+
+In this schema, the `userById` query uses the `@is` directive with {FieldSelectionMap} to declare that
+the `userId` argument is semantically equivalent to the `User.id` field. 
+
+An example query might look like this:
+
+```graphql
+query {
+    userById(userId: "123") {
+        id
+    }
+}
+```
+
+Here, it is exptected that the `userId` "123" corresponds directly to `User.id`, resulting in the
+following response if correctly implemented:
+
+```json
+{
+    "data": {
+        "userById": {
+            "id": "123"
+        }
+    }
+}
+```
+
+The {FieldSelectionMap} scalar type is used to establish semantic equivalence between an argument and
+the fields within the associated return type. To accomplish this, the scalar must define the
+relationship between input fields or arguments and the fields in the resulting object. 
+Consequently, a {FieldSelectionMap} only makes sense in the context of a specific argument and its 
+return type.
```

Comment:
I think the situation is different for `@require`. When you put `@require` on an argument, the mapping you're describing is not between the argument and the *return type* of the field, but between the argument and the `object type` the field is defined on. Take this example:

```graphql
type Product {
  id: ID!
  delivery(
    zip: String!
    size: Int! @require(field: "dimension { size }")
    weight: Int! @require(field: "dimension { weight }")
  ): DeliveryEstimates
}
```

In here, `"dimension { size }"` refers to a field of `Product`, not `DeliveryEstimates `.

I think this means we probably need to describe this in more neutral terms that apply both to the `@is` and the `@require` use cases. Perhaps we could state that a `{FieldSelectionMap}` always describes a mapping to a composite type, but that directives differ in how they establish that context.

Reply:
good catch, resolved